libimage/pull.go

```diff
@@ -65,36 +92,36 @@ func (r *Runtime) Pull(ctx context.Context, name string, pullPolicy config.PullP
 		// NOTE: the `docker` transport is an exception to support a
 		// `pull docker:latest` which would otherwise return an error.
 		if t := alltransports.TransportFromImageName(name); t != nil && t.Name() != registryTransport.Transport.Name() {
-			return nil, err
+			return "", nil, nil, err
 		}
 
 		// If the image clearly refers to a local one, we can look it up directly.
 		// In fact, we need to since they are not parseable.
 		if strings.HasPrefix(name, "sha256:") || (len(name) == 64 && !strings.ContainsAny(name, "/.:@")) {
 			if pullPolicy == config.PullPolicyAlways {
-				return nil, errors.Errorf("pull policy is always but image has been referred to by ID (%s)", name)
+				return "", nil, nil, errors.Errorf("pull policy is always but image has been referred to by ID (%s)", name)
 			}
-			local, _, err := r.LookupImage(name, nil)
+			local, resolvedName, err := r.LookupImage(name, nil)
 			if err != nil {
-				return nil, err
+				return "", nil, nil, err
 			}
-			return []*Image{local}, err
+			return resolvedName, ref, local, nil
```

Comment:
Is this `resolvedName, nil, local, nil`? AFAICS `ref` comes from the failing `ParseImageName` (and thus is, per the API, undefined).

Reply:
Good catch!  There's a (yet) unwritten contract that the returned image *must* be used if set.  Will make that clearer in the comments.